src/hakunapi-jsonfg/src/main/java/fi/nls/hakunapi/jsonfg/JSONFG.java

```diff
@@ -0,0 +1,132 @@
+    public static final String MIME_TYPE = MEDIA_MAIN_TYPE + "/" + MEDIA_SUB_TYPE;
+
+    public static final byte[] CONFORMS_TO = "conformsTo".getBytes();
+    public static final byte[] FEATURE_TYPE = "featureType".getBytes();
+    public static final byte[] GEOMETRY_DIMENSION = "geometryDimension".getBytes();
+    public static final byte[] FEATURE_SCHEMA = "featureSchema".getBytes();
+    public static final byte[] COORD_REF_SYS = "coordRefSys".getBytes();
+
+    public static final byte[] TIME = "time".getBytes();
+    public static final byte[] DATE = "date".getBytes();
+
+    public static final byte[] TIMESTAMP = "timestamp".getBytes();
+    public static final byte[] INTERVAL = "interval".getBytes();
+
+    public static final byte[] GEOMETRY = "geometry".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] PLACE = "place".getBytes();
+
+    public static final byte[][] CONF = new byte[][] {
+            //
+            "http://www.opengis.net/spec/json-fg-1/0.2/conf/core".getBytes(),
+            // "http://www.opengis.net/spec/json-fg-1/0.2/conf/types-schemas".getBytes()
+    };
+
+    /*
+     * TODO: JSONFG schema differs from JSON schema. We'll need JSONFG schema
+     * resource or SchemaFormatFactorySpi or such to support providing JSONFG schema
+     */
+    public static String getJSONFGSchema(FeatureType ft) {
+        return null;
+    }
+
+    public static Integer getGeometryDimension(HakunaPropertyGeometry geomType) {
+        if (geomType == null) {
+            return null;
+        }
+
+        Integer geometryDimension = null;
+
+        switch (geomType.getGeometryType()) {
+        case POINT:
+        case MULTIPOINT:
+            geometryDimension = 0;
+            break;
+        case LINESTRING:
+        case MULTILINESTRING:
+            geometryDimension = 1;
+            break;
+        case POLYGON:
+        case MULTIPOLYGON:
+            geometryDimension = 2;
+            break;
+        case GEOMETRY:
+        case GEOMETRYCOLLECTION:
+        default:
+            break;
+
+        }
+        return geometryDimension;
+    }
+
+    public static void writePlace(HakunaJsonWriter json, HakunaGeometry placeGeometry, HakunaGeometry geometry,
+            JSONFGGeometryWriter placeJson, GeometryWriter geometryJson, ProjectionTransformer outputCrs84Proj)
+            throws Exception {
+        if (placeGeometry != null) {
+
+            placeGeometry.write(placeJson);
+            if (geometry != null) {
+                geometry.write(geometryJson);
+            }
+        } else if (geometry != null) {
+            geometry.write(geometryJson);
+        } else {
+            json.writeFieldName(GEOMETRY);
+            json.writeNull();
+        }
+    }
+
+    public static void writeTemporal(HakunaJsonWriter json, LocalDate date, Instant timestamp) throws IOException {
+        if (date == null && timestamp == null) {
+            return;
```

Comment:
This is probably fine, hopefully JSONFG doesn't require `"time": null` if the value is null (like GeoJSON does with `"properties": null`)

Reply:
Actually looking into this it looks like `"time"` property is required:
https://docs.ogc.org/DRAFTS/21-045.html#schema-feature
but value can be either null or object:
https://docs.ogc.org/DRAFTS/21-045.html#schema-time

So my guess is that this is against the schema and we should replace the early return with with writing `"time": null`